src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1331,7 +1331,7 @@ public void visitVarDef(JCVariableDecl tree) {
                as these can undergo updates via copy on write.
             */
             if (tree.init != null) {
-                if ((v.flags_field & FINAL) == 0 || ((v.flags_field & STATIC) == 0 && v.owner.isPrimitiveClass()) ||
+                if ((v.flags_field & FINAL) == 0 || ((v.flags_field & STATIC) == 0 && (v.owner.isPrimitiveClass() || v.owner.isValueClass())) ||
                     !memberEnter.needsLazyConstValue(tree.init)) {
```

Comment:
There is still one instanceof isPrimitiveClass() || isValueClass() left in

Reply:
Oups. Good catch.